app/src/main/java/com/project/myapp/screens/auth/AuthActivity.kt

```diff
@@ -0,0 +1,220 @@
+package com.project.myapp.screens.auth
+
+import android.content.Intent
+import android.os.Bundle
+import android.util.Patterns
+import android.widget.Toast
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.app.ActivityOptionsCompat
+import androidx.core.widget.doOnTextChanged
+import com.project.myapp.R
+import com.project.myapp.databinding.ActivityAuthBinding
+import com.project.myapp.screens.main.MainActivity
+
+class AuthActivity : AppCompatActivity() {
+    private val binding: ActivityAuthBinding by lazy {
+        ActivityAuthBinding.inflate(layoutInflater)
+    }
+    private val viewModel: AuthViewModel by viewModels()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(binding.root)
+        setFocusListener()
+        setTextChangedListener()
+        setOnClickListener()
+    }
+
+    /* Processes register button
+    if email and password are correct - starts next activity and gives name from email
+    else shows errors
+     * */
+    private fun setOnClickListener() {
+        binding.apply {
+            buttonAuthRegister.setOnClickListener {
+                validateEmail(textInputEditTextAuthEmail.text)
+                validatePassword(textInputEditTextAuthPassword.text)
+                if (isEmailAndPasswordCorrect()) {
```

Comment:
it's odd to see that you validate email and password and then check their validity. I would rather have some link this:
```suggestion
                if (isEmailValid() && isPasswordValid()) {
```

Reply:
Also, it makes sense to keep in setOnClickListener just setting listeners and move all the click events to separate functions.

Reply:
I moved this logic to separated functions [ba705ab](https://github.com/cValentyna/myAndroidProject/pull/11/commits/ba705abf1603225de65a864cf10707dff8327df4), but after next movements it was a little bit changed in [424fa16](https://github.com/cValentyna/myAndroidProject/pull/11/commits/424fa1694f60903a312870565eb0d4f02883cfe7)